src/SelfUpdateCommand.php

```diff
@@ -41,6 +43,7 @@ protected function configure()
         $this
             ->setAliases(array('update'))
             ->setDescription("Updates $app to the latest version.")
+            ->addOption('allow-unstable', NULL, InputOption::VALUE_NONE, 'Allow unstable (e.g., alpha, beta, etc.) releases to be downloaded')
```

Comment:
Could we follow Composer's lead here and have --stable, --preview and --snapshot flags? It is not necessary to implement --snapshot as part of this PR; starting with just --stable and --preview would be sufficient.  (The --stable flag is the default and therefore a no-op.)

It would also be nice to have --<major-version> flags like Composer, but again, this is wish-list, not something required for this PR.